docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,350 @@
+.. _Courses API Blocks Resource:
+
+########################################
+Courses API Blocks Resource
+########################################
+
+With the Courses API **Blocks** resource, you can complete the
+following tasks.
+
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _Get a list of the course blocks in a course:
+
+*******************************************************
+Get a List of Course Blocks in a Specified Course
+*******************************************************
+
+The endpoint to get a list of course blocks in a specified block tree
+``/api/courses/v1/blocks/?course_id=<course_id>``.
+
+=====================
+Use Case
+=====================
+
+Get a list of course blocks in a specified course. Response results depend on
+the requesting user's access level.
+
+=====================
+Example Request
+=====================
+
+GET /api/courses/v1/blocks/?course_id=<course_id>
+
+GET /api/courses/v1/blocks/?course_id=edX%2FDemoX%2FDemo_Course
+&all_blocks=true&requested_fields=graded,format,student_view_multi_device
+
+=====================
+Parameters
+=====================
+
+This view redirects to /api/courses/v1/blocks/<root_usage_key>/ for the root
+block usage key of the course specified by course_id.
+
+This view accepts the following parameters.
+
+* block_counts: to return the aggregate number of blocks (including child
+  blocks) of each type specified in the "block_counts" response value.
+  Example: ``block_counts=video,problem``.
+
+* block_types_filter: Specify the types of blocks to be included in results.
+  Possible values are: ``sequential``, ``vertical``, ``html``, ``problem``,
+  ``video``   and ``discussion``. Example: ``block_types_filter=problem,html``.
+
+* course_id: (string, required) The ID of the course whose block data you are
+  requesting. Example: ``course_id=edX%2FDemoX%2FDemo_Course``.
+
+* depth: (integer or ``all``) Specify how far down in the course blocks
+  hierarchy to traverse down. A value of ``all`` specifies the entire
+  hierarchy. The default value is 0. Example: ``depth=all``.
+
+* requested_fields: (list) Indicates fields to return for each block, in
+  addition to id, type, and display_name, which are always returned. For the
+  list of possible fields, see the fields listed under ``blocks`` in the
+  **Response Values** section below. Example:
+  ``requested_fields=graded,format,student_view_multi_device``.
+
+* return_type: (string) Indicates in what data type to return the block data.
+  Supported values are ``dict`` and ``list``. The default value is ``dict``.
+
+* student_view_data: (list) Indicates for which block types the JSON data is
+  returned in the "student_view_data" response value. Example:
+  ``student_view_data=video``.
+
+* username: (string) Required, unless ``all_blocks`` is specified. Example:
```

Comment:
I think we should clarify how the user ID in the authentication token relates to the user ID in the username query parameter. If they are different, does the authentication token user need to have permission to query other users' courses? Why is the username query parameter required if the requesting user is known based on the authentication token?